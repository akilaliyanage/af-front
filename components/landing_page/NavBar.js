import React, { Component } from 'react'
import '../../assets/css/Landing_Page/index.css'
import config from '../../config.json'
import { Avatar, Image } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import {AiOutlineUser, AiOutlineDown } from 'react-icons/ai'
import { Link } from 'react-router-dom';
import { Tooltip } from 'antd';

class NavBar extends Component {
    constructor(props) {
        super(props);
        this.fetchTopic = this.fetchTopic.bind(this)
        this.logout = this.logout.bind(this);
        this.state = {
            navbar_items: [],
            topic : '',
            show : true
         }
    }

    componentDidMount(){
        this.fetchTopic()

    }

    fetchTopic(){
        fetch(config.host + "/topic").then(res => res.json()).then(data => {
            this.setState({topic : data[0]["topic"]})
        }).catch(err =>{
            //alert(err)
        })
    }

    logout(){
        window.localStorage.setItem('username','')
        window.localStorage.setItem('token','')
        window.localStorage.setItem('id','')
        window.localStorage.setItem('proImg','')
        window.localStorage.setItem('role','')

        window.location.replace('/home');
    }

    render() { 


        //const item = <li hidden={window.localStorage.getItem('token') == null?true : false} onClick={this.logout} style={{backgroundColor:'#1b4d72',color:'rgb(63, 65, 69)',padding:'10px', borderRadius:'10px'}} className="akila-menu-item"><a href=""><AiOutlineUser/>   hello {window.localStorage.getItem('username') + ' | SIGN-OUT'}</a></li>
        const drop = <div className="dropdown">
        <a><AiOutlineDown/> REGISTRATIONS</a>
            <div className="dropdown-content">
                <br />
                <Link to="/register"><p>Attendee </p></Link>
                <Link to=""><p>researcher Registration</p></Link>
                <Link to=""><p>workshop presenter</p></Link>
                <Link to="/editor/reg"><p>Editor</p></Link>
            </div>
        </div>

        const login = <div className="dropdown">
        <a><AiOutlineDown/> LOGINS</a>
            <div className="dropdown-content">
                <br />
                <Link to="/login"><p>Attendee Login</p></Link>
                <Link to="/researcher-login"><p>researcher Login</p></Link>
                <Link to=""><p>workshop Login</p></Link>
                <Link to="/admin/login"><p>Admin Login</p></Link>
                <Link to="/edi-login"><p>Editor Login</p></Link>
                <Link to="/reviewLogin"><p>Reviewer Login</p></Link>
            </div>
        </div>

        const user = <div className="dropdown" style={{zIndex:'3'}} onClick={this.logout}>
             <Tooltip title={window.localStorage.getItem('username') + " | Click here to logout"}>
             <Avatar src={window.localStorage.getItem('proImg')} />
             </Tooltip>
       
        </div>
        return (
            <div className='akila-header' data-testid='nav-bar'>
                <nav className='akila-navbar'>
                    <Link to="/home"><a href="" className='akila-brand'>{this.state.topic}</a></Link>
                    <input type="checkbox" id="nav" className='akila-hidden' />
                    <label htmlFor="nav" className='akila-nav-toggle'>
                        <span></span>
                        <span></span>
                        <span></span>
                    </label>

                    <div className='akila-wrapper'>
                        <ul className='akila-menu'>
                            {this.props.items.map(item => {
                                return (
                                    <li data-testid="nav-list" className="akila-menu-item" key={item._id}><a href={item.url}>{item.name}</a></li>
                                );
                            })}
                             <li hidden={this.state.show ? false : true} 
                             className="akila-menu-item"> 
                                {window.location.href.includes('/home') ? drop : null}
                             </li>
                             
                            <li className="akila-menu-item">
                                {login}
                            </li>

                            <li onClick={this.logout} className="akila-menu-item">
                                {user}
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>
         );
    }
}
 
export default NavBar;