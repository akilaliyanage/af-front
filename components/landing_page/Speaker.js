import React, { Component } from 'react'
class Speaker extends Component {
    constructor(props) {
        super(props);
        this.state = { 

         }
    }
    render() { 
        console.log(this.props.data);
        return (
            <div className="akila-speaker">
               <img data-testid="speaker-img" src={this.props.data.image} alt=""/>
               {this.props.colored?<canvas className="coveringCanvas"></canvas>:null}
                <div className="overlay">
                    <p>{this.props.data.name}</p>
                    <p>{this.props.data.des}</p>
               </div>
            </div>
         );
    }
}
 
export default Speaker;