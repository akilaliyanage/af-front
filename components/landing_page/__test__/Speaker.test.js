import React  from 'react';
import ReactDOM from 'react-dom';
import {render, screen, cleanup} from '@testing-library/react'
import NavBar from '../NavBar'
import { Router, Route } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';
import Speaker from '../Speaker'

//checking the componnet has a img tag

it('check the component has a img tag', () =>{
    render(
        <Speaker/>
)
    const todoElement = screen.getByTestId('speaker-img')
    //expect(getByTestId('nav-bar')).toContaineHTML('<li>')
    expectExport(todoElement).toContainHTML("<img")
})

//creating a snapshot
it('create a snapshot', () =>{
    const tree = renderer.create(
        <Speaker/>
    ).toJSON();

    expectExport(tree).toMatchSnapshot();
})
