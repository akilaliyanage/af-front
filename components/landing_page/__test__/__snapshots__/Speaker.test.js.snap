// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a snapshot 1`] = `
<div
  className="akila-speaker"
>
  <img
    alt=""
    data-testid="speaker-img"
  />
  <div
    className="overlay"
  >
    <p>
      Prof. Nobody
    </p>
    <p>
      SLIIT, Sri Lanka
    </p>
  </div>
</div>
`;
